### ISSUE TITLE
Channel pins in "environment.yaml" files are not saved to package specs

### ISSUE THREAD

**Issue by sdvillal** [conda/conda#6385]
Channel pins specified in environment files are not respected.  For example,run ```conda env create``` with this environment file:

```yaml
name: channel-not-written-to-user-specs

dependencies:
  - defaults::six
```

If we look at conda-meta/six*.json we can see that the channel pin has not been added to specs. Where we should read ```"requested_spec": "defaults::six"```, we only find ```"requested_spec": "six"```.

This is with conda 4.4.0rc2.

**Comment by sdvillal** [comment]
@kalefranz This keeps being problematic for us. At the moment, I workaround it by avoiding calling ```conda env create``` at all. Instead I parse the yaml file myself and call ```conda install``` - it is a very simple piece of code.

I think a similar approach could be used in the conda-env installer: remove most of that code and make it a very thin wrapper around conda create/install. Would that be the best approach to solve this and other future problem with conda-env not keeping up with changes? Perhaps I do not understand well and there is a reason why this code duplication needs to happen.

**Comment by dhirschfeld** [comment]
Hmm, I was hoping to use an `env.yml` file to work around my channel pinning issue #7177 but it looks like that won't work because of this issue :/

**Comment by sdvillal** [comment]
Should we add a test for this? I can try later after work.

**Comment by sdvillal** [comment]
@kalefranz  this bug is not fixed yet and so the issue should be reopened (just try the little example in the description).

**Comment by kalefranz** [comment]
It’s not fixed in 4.6? 4.6.0b0 is available on the conda-canary channel if you want to test.

**Comment by sdvillal** [comment]
Tested the beta version and it is indeed fixed there. I guess it could make sense to backport https://github.com/conda/conda/pull/7178?
